library/std/src/sys/pal/unix/fd.rs

```diff
@@ -529,9 +542,12 @@ impl FileDesc {
 
     pub fn get_access_mode(&self) -> io::Result<AccessMode> {
         let access_mode = self.get_flags()? & libc::O_ACCMODE;
-        Ok(AccessMode {
-            readable: access_mode == libc::O_RDWR || access_mode == libc::O_RDONLY,
-            writable: access_mode == libc::O_RDWR || access_mode == libc::O_WRONLY,
+        Ok(match access_mode {
+            libc::O_RDWR => AccessMode::ReadWrite,
+            libc::O_RDONLY => AccessMode::ReadOnly,
+            libc::O_WRONLY => AccessMode::WriteOnly,
+
+            _ => AccessMode::None,
```

Comment:
My thinking was to avoid this altogether. Is it ever actually reachable?

Reply:
According to the [`open`](https://man7.org/linux/man-pages/man2/open.2.html) it should not happen, however I think it should not panic upon any unexpected value.

I think we can fallback to read-only in that case?

Reply:
Returning an error seems the right thing to me.

Reply:
Replaced `TryFrom<Owned*>` with `From` to maintain consistency with `File`

https://github.com/rust-lang/rust/pull/127153#discussion_r1672170703